packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts

```diff
@@ -354,17 +376,19 @@ export function generateOperationMessagesForClients(
 				.slice(0, random.integer(1, 3));
 			applyMessages(toApply[0][0].sequenceNumber - 1, toApply, [client], logger);
 		}
-
 		const len = client.getLength();
 		const sg = client.peekPendingSegmentGroups();
 		let op: IMergeTreeOp | undefined;
 		if (len === 0 || len < minLength) {
-			const text = client.longClientId!.repeat(random.integer(1, 3));
-			op = client.insertTextLocal(random.integer(0, len), text);
+			op =
+				insertText === undefined ? generateInsert(client, random) : insertText(client, random);
 		} else {
 			let opIndex = random.integer(0, operations.length - 1);
-			const start = random.integer(0, len - 1);
-			const end = random.integer(start + 1, len);
+			// TODO: without accounting for potential fields here, we hit MergeTree insert failures.
```

Comment:
this doesn't really make sense to me, as each operation is field aware? i don't understand how the values generated here help? don't we just re-compute them every time within the operation itself?

Reply:
i'm a bit worried we are over constraining our positions here, and that the only reason it works, as i can't see why this is necessary